testing/test_stepwise.py

```diff
@@ -156,14 +156,18 @@ def test_change_testfile(stepwise_testdir):
     assert "test_success PASSED" in stdout
 
 
-def test_stop_on_collection_errors(broken_testdir):
-    result = broken_testdir.runpytest(
-        "-v",
-        "--strict-markers",
-        "--stepwise",
-        "working_testfile.py",
-        "broken_testfile.py",
-    )
-
-    stdout = result.stdout.str()
-    assert "errors during collection" in stdout
+@pytest.mark.parametrize("broken_first", [True, False])
+def test_stop_on_collection_errors(broken_testdir, broken_first):
+    """Stop during collection errors. We have two possible messages depending on the order (#5444),
+    so test both cases."""
+    files = ["working_testfile.py", "broken_testfile.py"]
+    if broken_first:
+        files.reverse()
+    result = broken_testdir.runpytest("-v", "--strict-markers", "--stepwise", *files)
+
+    if broken_first:
+        result.stdout.fnmatch_lines(
+            "*Error when collecting test, stopping test execution*"
```

Comment:
Why does the test order effect the output?

Reply:
When using pytest normally, the output is the same:

```
 λ pytest foo.py broken.py -q
============================== ERRORS ===============================
____________________ ERROR collecting broken.py _____________________
broken.py:1: in <module>
    foobar
E   NameError: name 'foobar' is not defined
!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!
1 error in 0.07 seconds
```

```
λ pytest broken.py foo.py -q
============================== ERRORS ===============================
____________________ ERROR collecting broken.py _____________________
broken.py:1: in <module>
    foobar
E   NameError: name 'foobar' is not defined
!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!
1 error in 0.07 seconds
```

But `--sw` changes that: it sees a failure and immediately stops the execution itself, rather than letting pytest itself normally do it.

Hmmm it seems that `sw` implemented this before pytest itself adopted the behavior of "stop the test session during any test collection". I think we can remove the special status in this case anyway.

Reply:
Fixed, thanks for bringing that up. 👍